src/auth/externalAccountAuthorizedUserClient.ts

```diff
@@ -309,6 +312,46 @@ export class ExternalAccountAuthorizedUserClient extends AuthClient {
       this.refreshToken = refreshResponse.refresh_token;
     }
 
+    // Set credentials and refresh trust boundary data.
+    this.credentials = {...this.cachedAccessToken};
+    delete (this.credentials as CredentialsWithResponse).res;
+    this.trustBoundary = await this.refreshTrustBoundary(this.credentials);
+
     return this.cachedAccessToken;
   }
+
+  /**
+   * Retrieves the workforce pool ID from the audience.
+   *
+   * The audience has the format:
+   * `//iam.googleapis.com/locations/global/workforcePools/{pool_id}/providers/{provider_id}`
+   *
+   * @returns The workforce pool ID, or `null` if it cannot be determined.
+   */
+  getWorkforcePoolId(): string | null {
```

Comment:
Looks like this is duplicated, can we create a shared utility function so both BaseExternalAccountClient and ExternalAccountAuthorizedUserClient could use it?